### ISSUE TITLE
Windows: __declspec(dllexport) even when trying to import DLL

### ISSUE THREAD

**Issue by imi0** [eclipse-paho/paho.mqtt.c#742]
Under Windows, all headers unconditional declare exported functions as __declspec(dllexport).

This means, we can not use the same header files when trying to use the library (because then, it must be "__declspec(dllimport)") and there are no headers that use the correct import declaration :(.

The current way is a bit weird to my eyes anyway.. Usually, you do something like this:

```
#ifdef _DLL
  #ifdef PAHO_DLLEXPORT
    #define PAHO_EXPORT __declspec(dllexport)
  #else
    #define PAHO_EXPORT __declspec(dllimport)
  #endif
#else
  #define PAHO_EXPORT
#endif

...

PAHO_EXPORT void myfunction();
...
```

And then define "_DLL" when you want to build as a dll and define "PAHO_DLLEXPORT" when you build Paho.dll itself (and don't define it when you just importing the dll and its headers in another project)

**Comment by icraggs** [comment]
The library was intended to be built as a DLL - I hadn't given it any thought about being built any other way.

**Comment by icraggs** [comment]
I think this has been fixed in the develop branch by a recent PR #341

**Comment by keysight-daryl** [comment]
It actually won't work not as a dll under windows due to the DllMain being the place where all of the global mutexes are initialized.
